### PR TITLE
SONAR-6380 add Java WS to uninstall a plugin

### PR DESCRIPTION
**Pull request by sns-seb**
add a Java WS to uninstall an installed plugin

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      .setDescription("The key identifying the plugin to uninstall")
+      .setRequired(true);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    String key = request.mandatoryParam(PARAM_KEY);
+    ensurePluginIsInstalled(key);
+    pluginJarsInstaller.uninstall(key);
+  }
```
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java]: There is no response to this WS. In this case, you should call response.noContent() to have a correct 204 status code.

**Comment by jblievremont** [comment]
Just some tiny feedback, then LGTM!

**Comment by sns-seb** [comment]
applied feedback and rebased on master

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java]: you're right, I'll also add an assertion in the unit test making sure the response is empty (unfortunately, it does not seem possible to verify the HTTP status with `TestResponse`)

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller.createAction("uninstall")
+      .setPost(true)
+      .setDescription("Uninstalls the plugin specified by its key.")
+      .setHandler(this);
+
+    action.createParam(PARAM_KEY)
+      .setDescription("The key identifying the plugin to uninstall")
+      .setRequired(true);
```
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java]: Last minor feedback: can you provide an example value?

Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java]: initially thought about providing an example value, but for a plugin key, I didn't see the value

File: server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java]: Security check - user must be a system administrator.

**Comment by simonbrandhof** [comment]
Added a blocker comment.

**Comment by sns-seb** [comment]
PR updated to take blocker comment into account and rebase on master

**Comment by simonbrandhof** [comment]
Added comment about description, then LGTM
